### PR TITLE
Feature/3545  allow user to leave their own space

### PR DESCRIPTION
**Pull request by asf-algoo**
related to #3545

## Checkpoints

**For developers**

- [x] If relevant, manual tests have been done to ensure the stability of the whole application and that the involved feature works
- [x] The original issue is up to date w.r.t the latest discussions and contains a short summary of the implemented solution
- [x] Automated tests covering the feature or the fix, have been written, deemed irrelevant (give the reason), or an issue has been created to implement the test (give the link)

**For code reviewers**

- [x] The code is clear enough
- [x] If there are FIXMEs in the code, related issues are mentioned in the FIXME
- [x] If there are TODOs, NOTEs or HACKs in code, the date and the developer initials are present

**For testers**

- [x] Manual, quality tests have been done

### CODE REVIEW

File: backend/tracim_backend/lib/core/userworkspace.py [hunk 1]
```diff
@@ -147,22 +156,42 @@ def create_one(
...
+        # INFO - G.M - 2020-17-09 - check if user is workspace_manager of workspace
+        try:
+            self.get_all_for_workspace_query(workspace_id).filter(
+                UserRoleInWorkspace.role == WorkspaceRoles.WORKSPACE_MANAGER.level
+            ).filter(UserRoleInWorkspace.user_id == user_id).one()
+        except NoResultFound:
+            return False
+
+        # INFO - G.M - 2020-17-09 - check other potential workspace_manager of workspace
+        return not bool(
```
Comment (grignards) [inline backend/tracim_backend/lib/core/userworkspace.py:178]: Wouldn't `return ....count()>0` be more readable ?

**Review by grignards — APPROVED** [review]

File: backend/tracim_backend/lib/core/userworkspace.py (hunk 1 above)
Comment (inkhey) [reply backend/tracim_backend/lib/core/userworkspace.py:178]: Right that's more obvious.

Comment (inkhey) [reply backend/tracim_backend/lib/core/userworkspace.py:178]: in fact, it should be `count() == 0`

Comment (inkhey) [reply backend/tracim_backend/lib/core/userworkspace.py]: fixed

**Review by raphj — APPROVED** [review]

**Comment by inkhey** [comment]
About testing this:
- current frontend does not allow now to remove user from his own space, so, this part can also be checked manually through api. I suggest to way for frontend to test it
- New limitation disallowing to remove user from space if last space manager can be tested through frontend : create a space with one user, try changing the role or removing it from space using another admin account.

**Comment by gbonaspetti** [comment]
Tested on branch.

**Pull request merged**